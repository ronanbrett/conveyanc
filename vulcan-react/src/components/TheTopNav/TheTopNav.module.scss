@import "src/styles/vars";

.TheTopNav {
  grid-area: header;

  header {
    width: 100%;
    height: 100%;
    background-color: var(--white-color);
    border-top-left-radius: 45px;

    padding: 0 var(--spacing-m) 0 var(--spacing-xl);
    display: flex;
    align-items: center;

    border-bottom: 1px solid var(--border-color);
  }

  h1 {
    @include type-header-s;
    margin: 5px var(--spacing-mx) 0 0;
  }

  h2 {
    @include type-header-s;
    margin: 5px var(--spacing-l) 0 var(--spacing-mx);
  }
}
