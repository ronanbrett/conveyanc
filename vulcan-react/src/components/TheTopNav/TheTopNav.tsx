import React from "react";
import styles from "./TheTopNav.module.scss"; // Import css modules stylesheet as styles

import { useLocation } from "react-router-dom";
import { getRouteMeta } from "core/routes/routes.config";

interface TheTopNavProps {
  children?: any;
}

const TheTopNav = (props: TheTopNavProps) => {
  // const { pathname } = useLocation();
  // const route = getRouteMeta(pathname);

  return (
    <div data-testid="TheTopNav" className={styles.TheTopNav}>
      <header>
        {/* <h1>{route.title}</h1> */}
        <div className="vertical-spacer vertical-spacer--half"></div>
        <h2 className="subTitle">Subtitle</h2>
      </header>
    </div>
  );
};

export default TheTopNav;
