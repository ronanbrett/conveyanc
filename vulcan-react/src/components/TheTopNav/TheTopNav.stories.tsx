import React from "react";

import TheTopNav from "./TheTopNav";

import { action } from "@storybook/addon-actions";

export default {
  title: "Components/TheTopNav",
  decorators: [],
  component: TheTopNav,
};

export const Default = () => <TheTopNav></TheTopNav>;
