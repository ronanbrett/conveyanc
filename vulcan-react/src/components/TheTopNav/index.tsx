import TheTopNav from './TheTopNav';

export default TheTopNav;