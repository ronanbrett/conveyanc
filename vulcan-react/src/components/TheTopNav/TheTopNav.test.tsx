import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React from "react";
import TheTopNav from "./TheTopNav";
import { Router, Route } from "react-router-dom";

const history = createMemoryHistory({ initialEntries: ["/"] });

const TestComponent = () => (
  <Router history={createMemoryHistory({ initialEntries: ["/"] })}>
    <Route path="/" component={() => <TheTopNav />} />
  </Router>
);

describe("TheTopNav", () => {
  it("renders without error", () => {
    const { asFragment } = render(<TestComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
