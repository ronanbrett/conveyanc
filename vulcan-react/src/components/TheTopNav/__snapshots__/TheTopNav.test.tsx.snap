// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheTopNav renders without error 1`] = `
<DocumentFragment>
  <div
    class="TheTopNav"
    data-testid="TheTopNav"
  >
    <header>
      <div
        class="title"
      >
        <h1>
          Home
        </h1>
      </div>
      <div
        id="topNavPortal"
      />
    </header>
  </div>
</DocumentFragment>
`;
